docs/docs/general/03-Guides/reporter.md

```diff
@@ -0,0 +1,109 @@
+
+Also [check out the repository](https://github.com/askui/askui-reporters) for more details.
+
+## Installation
+Install `@askui/askui-reporters` as a dev-dependency:
+
+```bash
+npm install --save-dev @askui/askui-reporters
+```
+
+## Usage of Allure-Reporter
+Add the reporter to the `UiControlClient` in `jest.setup.ts`:
+
+```typescript
+// Do not forget this import at the start of the file
+import { AskUIAllureStepReporter } from "@askui/askui-reporters";
+...
+  aui = await UiControlClient.build({
+    reporter: new AskUIAllureStepReporter({
+      withScreenshots: 'always' as const, // See below for possible values
+      withDetectedElements: 'always' as const, // See below for possible values
+    })
+  });
+...
+```
+
+You can pass a `ReporterConfig` object to the reporter to configure the level of detail for screenshots and detected elements. The default values are `'onFailure'` for both:
+
+```typescript
+/**
+ * SnapshotDetailLevel represents various levels of detail for the snapshot.
+ * There are four possible values for this type.
+ *
+ * @typedef {'required'|'onFailure'|'begin'|'always'} SnapshotDetailLevel
+ *
+ * @property {'required'} required - Snapshot details, like screenshots or detected elements, may be available if required by the step. However, their presence is not guaranteed.
+ * @property {'onFailure'} onFailure - Snapshot details are available when the step fails, primarily for debugging purposes. This level includes everything provided by 'required'.
+ * @property {'begin'} begin - Snapshot details are available when the command starts, useful for determining why certain elements were interacted with. This level includes everything provided by 'onFailure'.
+ * @property {'always'} always - Snapshot details are consistently available, irrespective of whether a step has failed or not, aiding in debugging. 
+ */
+
+/**
+ * The ReporterConfig interface encapsulates the configuration options for the reporter.
+ *
+ * @interface ReporterConfig
+ *
+ * @property {SnapshotDetailLevel} [withScreenshots='onFailure'] - Defines the detail level for screenshots. Acceptable values: 'required', 'onFailure', 'begin', 'always'. Note: Higher levels of detail may impede step execution speed.
+ * @property {SnapshotDetailLevel} [withDetectedElements='onFailure'] - Defines the detail level for detecting elements. Acceptable values: 'required', 'onFailure', 'begin', 'always'. Note: Higher levels of detail may impede step execution speed and incur additional costs.
+ */
+```
+
+#### Configure `beforeEach()` and `afterEach()` in `jest.setup.ts`
+The `UiControlClient` retrieves the videos and images from your `UiController`. You have to implement `beforeEach()` and `afterEach()` in `jest.setup.ts` to start the recording and then add it to your report:
+
+1. Allure Reporter
+```typescript
+// Do not forget this import at the start of the file
+import "jest-allure-circus";
```

Comment:
Without explanation what this is and why it is necessary but I think we already explained it in the reporters repo, maybe we can integrate that. 

Is it installed with `@askui/askui-reporters`?

Reply:
Yes, it is installed with `askui-reporters` automatically because I declared it as a dependency there (not a dev dependency).